sim/pokemon.ts

```diff
@@ -525,6 +525,22 @@ export class Pokemon {
 		return null;
 	}
 
+	allies(adjacentOnly?: boolean): Pokemon[] {
+		let allies = this.side.active;
+		if (this.side.ally) allies = allies.concat(this.side.ally.active);
```

Comment:
```
if (this.battle.gameMode === 'multi') {
  const team = this.side.n % 2;
  allies = this.battle.sides.flatMap(side => 
    side.n % 2 === team ? side.active : []
  );
}
```

n.b. this will require polyfilling `flatMap`, but that's totally worth it.

Reply:
Where would I do the polyfill? Just at the beginning of the file, or...?

Reply:
I'd put it in `Dex`, the Traditional Polyfill Location.

Reply:
could we just include it as a dependency? https://github.com/jonathantneal/array-flat-polyfill#usage

Reply:
Actually, can’t we do this?
```
if (this.gameType === 'multi') {
   const team = this.side.n % 2;
   for (const side of sides) {
      if (side !== this.side && side.n % 2 === team) {
         allies = allies.concat(side.active);
      }
   }
}
```
I think this is definitely possible without polyfills.

Reply:
Better yet:
```
if (this.gameType === 'multi') {
   const team = this.side.n % 2;
   allies = this.battle.sides.reduce((total, side) => side.n % 2 === team ? total.concat(side.active) : left, [] as Pokemon[]);
}

Reply:
I hate dependencies, especially when they're optimized for correctness in use cases I never use, rather than perf.

Reply:
I find `reduce` to be unreadable when used for things that aren't actually reductions, like flatmaps.

I'd rather use your first solution if you don't want to polyfill `flatMap`, but I'm confused why not.

Reply:
```
Array.prototype.flatMap = function (callback, thisArg) {
  const newArray = [];
  for (let i = 0; i < this.length; i++) {
    newArray.push(...callback.call(thisArg, this[i], i, this));
  }
  return newArray;
};
```

Stick this in Dex somewhere, that's all you need.

Reply:
I didn't want to polyfill because I didn't know how and every version I could find online was super complicated.

Reply:
The online versions tend to have features no one cares about. Just use this one.